include/picongpu/particles/debyeLength/Check.hpp

```diff
@@ -0,0 +1,180 @@
+                        1,
+                        pmacc::mpi::reduceMethods::AllReduce());
+                    reduce(
+                        pmacc::nvidia::functors::Add(),
+                        &globalResult.numActiveSupercells,
+                        &localResult.numActiveSupercells,
+                        1,
+                        pmacc::mpi::reduceMethods::AllReduce());
+                    if(globalResult.numActiveSupercells)
+                    {
+                        auto const ratioFailing = static_cast<float_64>(globalResult.numViolatingSupercells)
+                            / static_cast<float_64>(globalResult.numActiveSupercells);
+                        /* Due to random nature of temperature and estimates used internally,
+                         * we allow a relatively small number of supercells to fail the check
+                         */
+                        auto const thresholdFailing = 0.1;
+                        if(ratioFailing <= thresholdFailing)
+                            return Status::Passed;
+                        else
+                            return Status::Failed;
+                    }
+                    else
+                        return Status::Skipped;
+                }
+
+            } // namespace detail
+
+            /** Check Debye length resolution
+             *
+             * The check is currently done only for species called "e" and is supposed
+             * to be called just after the particles are initialized as start of a
+             * simulation. The results are output to log< picLog::PHYSICS >.
+             *
+             * This function must be called from all MPI ranks.
+             *
+             * @param cellDescription mapping for kernels
+             */
+            HINLINE void check(MappingDesc const cellDescription)
+            {
+                bool isPrinting = (Environment<simDim>::get().GridController().getGlobalRank() == 0);
+                using ElectronSpecies = pmacc::particles::meta::FindByNameOrType_t<
+                    VectorAllSpecies,
+                    PMACC_CSTRING("e"),
+                    pmacc::errorHandlerPolicies::ReturnType<void>>;
+                bool isElectronsFound = !std::is_same<ElectronSpecies, void>::value;
+                if(isElectronsFound)
+                {
+                    auto status = detail::checkDebyeLength<ElectronSpecies>(cellDescription);
+                    if(isPrinting)
+                        switch(status)
+                        {
+                        case detail::Status::Passed:
+                            log<picLog::PHYSICS>("Estimated Debye length for species \"e\" is resolved by the grid");
+                            break;
+                        case detail::Status::Failed:
+                            log<picLog::PHYSICS>(
+                                "Warning: estimated Debye length for species \"e\" is not resolved by the grid\n"
```

Comment:
Could we please output the calculated temperature and the required? In case the user is getting a `warning` he can see how much he/she is off.

Reply:
The temperature does not have to be constant. So it's not clear which value to output.

Reply:
The goal of this check how I see it is to say to a user "perhaps you forgot to set the temperature or set it incorrectly". I don't see an easy way to point directly at what's wrong. We can (and should regardless of this PR) however extend the documentation on which things should be resolved.